### ISSUE TITLE
Some intellisense don't work well in interactive window

### ISSUE THREAD

**Issue by linette-zyy** [microsoft/PTVS#6943]
**Environment**
![image](https://user-images.githubusercontent.com/59192491/160791099-5a8b3344-8966-49b1-855f-521e7316de86.png)

**Steps to Reproduce**
1.Create a new Python Application
2.Right click project and select Open Interactive Window
3.Write some codes in Interactive Window ,such as 'imp' to get 'import', 'de' to get 'def',  'pri' to get 'print'

**Expected behavior**
All IntelliSense should work well

**Actual behavior**
IntelliSense only work well when i first wrote code

**Additional context and screenshots**
![interactive_completion](https://user-images.githubusercontent.com/59192491/160794865-84e8bea4-c6ca-4f07-93ba-37a73ae1ad75.gif)

**Comment by StellaHuang95** [comment]
The bug does not exist in Microsoft Visual Studio Enterprise 2022 (64-bit) - Current Version 17.1.0, but does exist in Microsoft Visual Studio Enterprise 2022 (64-bit) - Int Preview Version 17.2.0 Preview 3.0 [32313.70.main].  Needs further investigation.

**Comment by StellaHuang95** [comment]
The `HandleTypedChar` in [this file](https://github.com/microsoft/PTVS/blob/18804814ac579d039058d79f6aa47890c86f0885/Python/Product/PythonTools/PythonTools/Repl/Completion/AsyncCompletionSource.cs#L181) is not invoked when typing second line in the interactive window.

**Comment by StellaHuang95** [comment]
Followed up with Rich, he suggested ask Christian Gunderman. This has to do with the core calling us when typing in a projection buffer. If it used to work, they likely changed something there.

**Comment by gundermanc** [comment]
@StellaHuang95 I'm not aware of any changes that would have specifically broken this, though Amadeusz did add support for projection to Async Completion in ~January: https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform/pullrequest/375941

**Comment by StellaHuang95** [comment]
An internal bug has been created here to track this issue: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1528388/

**Comment by StellaHuang95** [comment]
Will be fixed in 17.3 Preview 3.

**Comment by AmadeusW** [comment]
This looks like an issue in `Microsoft.PythonTools.Repl.Completion.AsyncCompletionSource`:
In this situation, I got completion at the first line (both `import` and `sys` were completed)
![image](https://user-images.githubusercontent.com/1673956/175709119-abce92ea-a702-4da7-b687-75c58e18d65d.png)
On the second line, however, I see that `Microsoft.PythonTools.Repl.Completion.AsyncCompletionSource.GetApplicableToSpan` calls `Microsoft.PythonTools.Repl.Completion.Utilities.ToSnapshotSpan` which calls `GetLineFromLineNumber(1)`, presumably to get the second line of code (contains `im`)
This fails because the `ITextSnapshot` contains only one line. These are the properties of the snapshot:
```
{version: 3 lines: 1 length: 2 
 content: im}
    Content: {im}
    ContentType: {Python}
    Length: 2
    LineCount: 1
    Lines: {Microsoft.VisualStudio.Text.Implementation.BaseSnapshot.<get_Lines>d__38}
    TextBuffer: {Python:"im"}
    TextBufferHelper: {Python:"im"}
    TextImage: {Microsoft.VisualStudio.Text.Implementation.CachingTextImage}
    Version: {V3 (r3)}
    cachingContent: {Microsoft.VisualStudio.Text.Implementation.CachingTextImage}
    contentType: {Python}
    textBuffer: {Python:"im"}
    version: {V3 (r3)}
```
@StellaHuang95 please investigate
